src/main/java/de/tum/in/www1/artemis/service/ConflictingResultService.java

```diff
@@ -0,0 +1,84 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.Feedback;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.modeling.ConflictingResult;
+import de.tum.in.www1.artemis.domain.modeling.ModelAssessmentConflict;
+import de.tum.in.www1.artemis.repository.ConflictingResultRepository;
+
+@Service
+public class ConflictingResultService {
+
+    private final Logger log = LoggerFactory.getLogger(ConflictingResultService.class);
+
+    private ConflictingResultRepository conflictingResultRepository;
+
+    public ConflictingResultService(ConflictingResultRepository conflictingResultRepository) {
+        this.conflictingResultRepository = conflictingResultRepository;
+    }
+
+    public ConflictingResult createConflictingResult(ModelAssessmentConflict conflict, Feedback feedback) {
+        ConflictingResult conflictingResult = new ConflictingResult();
+        conflictingResult.setModelElementId(feedback.getReferenceElementId());
+        conflictingResult.setResult(feedback.getResult());
+        conflictingResult.setConflict(conflict);
+        return conflictingResult;
+    }
+
+    public ConflictingResult createConflictingResult(String modelElementID, Result result) {
+        ConflictingResult conflictingResult = new ConflictingResult();
+        conflictingResult.setModelElementId(modelElementID);
+        conflictingResult.setResult(result);
+        // conflictingResult.setConflict(conflict);
+        return conflictingResult;
+    }
+
+    /**
+     * Removes
+     *
+     * @param conflict     the conflict object to update
+     * @param newFeedbacks represents the current feedbacks that the given conflict object is in conflict with
+     */
+    public void updateExistingConflictingResults(ModelAssessmentConflict conflict, List<Feedback> newFeedbacks) {
+        removeRemovedConflictingResults(conflict, newFeedbacks);
+        addMissingConflictingResults(conflict, newFeedbacks);
+    }
+
+    @Transactional
+    void removeRemovedConflictingResults(ModelAssessmentConflict conflict, List<Feedback> newFeedbacks) {
+        Set<ConflictingResult> existingConflictingResultsCopy = new HashSet<>(conflict.getResultsInConflict());
+        conflict.getResultsInConflict().clear();
+        Set<String> newFeedbacksElementIds = new HashSet<>();
```

Comment:
I would try to reduce the code here and in the other methods a bit, e.g. the assignment could be directly made if a map and then a collect is used:
Set<String> newFeedbacksElementIds = newFeedbacks.map(...).collect(Collectors.toSet()).
Also I don't think that conflict.getResultsInConflict() would need to be copied into another variable as stream().filter doesn't mutate the underlying data structure.

Reply:
Since Hibernate throws errors when a conflict gets assigned a new Instance of an ConflictingResult Set, i need the copy of resultsInConflicts. I did some some small adaptions to the methods based on ur feedback